docs: adds lexicon of terms and FAQ cheat sheet (#29)
complytime/complytime-policies: docs/FAQ.md

# Frequently Asked Questions

> **Question:** If I am using Gemara, do I need to know CUE?

Reviewer: Direct link to CUE would be great.

Author: Addressed -- CUE is now linked on first use: `[CUE](https://cuelang.org/)` (FAQ.md line 3).

**Answer:** Nope. CUE is for expressing the schemas and for validation. All you need to know is YAML.

**Question:** Is ComplyTime a Product?
**Answer:** No. Not at this time at least. ComplyTime is a platform of tools that are interoperable with one another. When we say "ComplyTime" we are talking about the ecosystem of interoperable tools that are used for continuous monitoring, scanning of workspace configurations using bundled OCI artifacts (e.g., Gemara Policies and Catalogs), and results that can be leveraged for a full decision and evaluation log to produce compliance report.

**Question:** How does complyctl work?
**Answer:** Check out the [complyctl section](complyctl-overview.md). Starting by fetching policies from the OCI registry, resolving dependency graphs, dispatch to providers, produce compliance reports

**Question:** What are providers?
**Answer:** See [complytime-providers](complytime-providers-overview.md) for more details.

**Question:** What is the difference between capital "P" Policy and lowercase "p" policy?

> **Answer:** Capital "P" Policy describes **rules** and what needs to be assessed in **natural language**. Lowercase "p" policy is the **operational logic** to evaluate structured input in **query language**. 

Reviewer: Ideally, we'd use a different term then. The difference isn't obvious when speaking.

**Question:** What is the Gemara-equivalent to cross-walking?
**Answer:** The Mapping Document Schema is the logical equivalent to cross-walking in the Gemara Model. A regulation like the European Union Cyber Resilience Act can overlap with other frameworks and standards like ISO 27001, etc. The overlap can be expressed in a Gemara MappingDocument to describe the relationships between the two frameworks.

**Question:** What is a Policy Engine? 

> **Answer:** the complytime-providers are examples of policy engines like AMPEL and OPA. They are decision makers. The policy engine interprets "policies" (e.g., checks or rules) and determines whether specific actions or behaviors comply with them. The providers essentially take the information that is stored and generated and determine how to evaluate whether the policy _IS_ or _IS NOT_ being satisfied. 

Reviewer: How do I integrate my own (can I)?

Author: Addressed -- added FAQ entry: "Can I integrate my own policy engine?" with details on the provider plugin interface (FAQ.md line 36-37).

**Question:** How does OPA work?

> **Answer:** In three steps, OPA uses **Query**, **Data + Policy**, and **Decision**. The **query** would be a question like "Can USER Camille delete this file?" OPA looks at **Policy (rules)** and the **Data (info about Camille and the file)**. Then OPA sends back a **decision** "Yes" or "No" (or even complex responses like "Yes, but only if she's using the VPN").

Reviewer: Does this work in a non-deterministic system like `polkit` where the exact state of the system determines the outcome?

**Question:** What is a policy (lowercase "p")?
**Answer:** A policy is the **operational logic** — code written in a query language (e.g., [Rego](COMMON-TERMS.md#rego)) — that a [policy engine](COMMON-TERMS.md#opa-open-policy-agent) evaluates against structured input to produce a pass/fail decision.

**Question:** What is the Gemara-equivalent to cross-walking?
**Answer:** The Mapping Document Schema is the logical equivalent to cross-walking in the Gemara Model. A regulation like the European Union Cyber Resilience Act can overlap with other frameworks and standards like ISO 27001, etc. The overlap can be expressed in a Gemara MappingDocument to describe the relationships between the two frameworks.

**Question:** What is a Policy Engine?
**Answer:** The complytime-providers are examples of policy engines like AMPEL and OPA. They are decision makers. The policy engine interprets "policies" (e.g., checks or rules) and determines whether specific actions or behaviors comply with them. The providers essentially take the information that is stored and generated and determine how to evaluate whether the policy _IS_ or _IS NOT_ being satisfied.

> **Question:** How does OPA work?

Reviewer 2: Do we really need to answer this question here?

Author: It was more from the original perspective of the Compliance Team. I'll remove it though since it's probably covered.

# Frequently Asked Questions

**Question:** If I am using Gemara, do I need to know [CUE](https://cuelang.org/)?
**Answer:** Nope. CUE is for expressing the schemas and for validation. All you need to know is YAML.

> **Question:** Is ComplyTime a Product?

Reviewer 2: Nit: I am unsure whether this question belongs here. 
```suggestion
```

**Answer:** A policy is the **operational logic** — code written in a query language (e.g., [Rego](COMMON-TERMS.md#rego)) — that a [policy engine](COMMON-TERMS.md#opa-open-policy-agent) evaluates against structured input to produce a pass/fail decision.

**Question:** What is the Gemara-equivalent to cross-walking?
**Answer:** The Mapping Document Schema is the logical equivalent to cross-walking in the Gemara Model. A regulation like the European Union Cyber Resilience Act can overlap with other frameworks and standards like ISO 27001, etc. The overlap can be expressed in a Gemara MappingDocument to describe the relationships between the two frameworks.

**Question:** What is a Policy Engine?
**Answer:** The complytime-providers are examples of policy engines like AMPEL and OPA. They are decision makers. The policy engine interprets "policies" (e.g., checks or rules) and determines whether specific actions or behaviors comply with them. The providers essentially take the information that is stored and generated and determine how to evaluate whether the policy _IS_ or _IS NOT_ being satisfied.

**Question:** My auditor wants a format that complyctl doesn't support. What do I do?

> **Answer:** complyctl currently supports [four output formats](complyctl-overview.md#for-audit-program-managers-evidence-collection): OSCAL assessment-results (JSON), SARIF, Markdown, and EvaluationLog (YAML). If your auditor requires a different format, you can post-process the OSCAL or EvaluationLog output — both are structured and machine-readable. OSCAL in particular is the NIST standard and has a growing ecosystem of conversion tools.

Reviewer 2: ```suggestion
**Answer:** complyctl currently supports [four output formats](complyctl-overview.md#for-audit-program-managers-evidence-collection): OSCAL assessment-results (JSON), SARIF, Markdown, and EvaluationLog (YAML). If your auditor requires a different format, you can post-process the OSCAL or EvaluationLog output — both are structured and machine-readable. Gemara and OSCAL have a growing ecosystem of conversion tools.
```

**Question:** Does Layer 3 require Layer 2 to be completed first?
**Answer:** No. Layer 3 compliance processes do not have a hard requirement to depend on Layer 2. However, Layer 3 does require at least Layer 1 or Layer 2 to be completed as a prerequisite. In other words, you can go from Layer 1 directly to Layer 3 without completing Layer 2, but you cannot start Layer 3 without any prior layer work.

**Question:** Are the boundaries between compliance layers rigid?
**Answer:** No. The boundaries between compliance layers are "soft" and iterative rather than rigid. Teams can move fluidly between layers as their compliance program matures, revisiting and refining earlier layers as needed.

**Question:** What is the difference between Layer 2 and Layer 3?
**Answer:** Layer 2 is technology-based. If a piece of guidance is technology-specific and you need to write a Rego policy for it, that is the trigger to use a Layer 2 control catalog. Layer 3 is for overriding assessment requirements. It allows a team to tighten constraints when adapting a standard commercial control catalog to a more stringent environment (such as FedRAMP) without needing to rewrite the entire catalog.

> ---

Author: @jpower432 Added FAQ entries from the discussion with Fabian.
